[WIP] PR to upgrade react-native Android side gradually

siddarthkay opened the pull request:
This PR uses https://react-native-community.github.io/upgrade-helper as a guideline and upgrades the react-native version Android side only.

We go step by step so that we resolve issues as and when they come in each step.
It was cumbersome process but It got us this far 😉

status: wip

siddarthkay committed:
upgrade react-js to 17.0.1

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8aff6a9b | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14708/1/) | 2023-01-04 13:54:29 | ~3 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-230104-135050-8aff6a-pr14708-tests.log) |

siddarthkay committed:
update flowconfig & prettierrc

siddarthkay committed:
bump jest and react-test-renderer

siddarthkay committed:
bump android to 0.64.4

siddarthkay committed:
bump react dependencies

as per https://react-native-community.github.io/upgrade-helper/?from=0.64.4&to=0.65.3
bumping
react js to 17.0.2
react-native-codegen to ^0.0.7
react-test-renderer to 17.0.2

siddarthkay committed:
update partial android changes for 0.64.4

As per : https://react-native-community.github.io/upgrade-helper/?from=0.63.5&to=0.64.4
updated the following files :
- android/app/build.grade
- android/app/src/debug/AndroidManifest.xml
- android/app/src/main/AndroidManifest.xml

siddarthkay committed:
take gradle plugin to 4.1.0

updated the version in gradle.properties
updated the version in nix/pkgs/aapt2/default.nix and added the osx hashes

siddarthkay committed:
updated hashes for linux

flexsurfer commented:
feel free to reopen if needed

jakubgs commented:
This PR needs to be rebased. The iOS build will forever be stuck in the queue because it expects old Xcode which doesn't exist:
https://github.com/status-im/status-mobile/blob/29a768c8aaf67d4b3f3cabc1e2ec075b058c84ca/ci/Jenkinsfile.ios#L7

siddarthkay commented:
https://stackoverflow.com/questions/68387270/android-studio-error-installed-build-tools-revision-31-0-0-is-corrupted

siddarthkay committed:
upgrade Gradle version to 6 now works

siddarthkay committed:
fix errors due to missing icons

siddarthkay committed:
updating .flowconfig

siddarthkay committed:
upgrade flipper version

siddarthkay committed:
upgrade react-native version to 0.65.3

com.facebook.react.ReactFragmentActivity was deprecated here :
https://github.com/facebook/react-native/commit/2798e7172b01b9e2dbe2937d0163f98ab29230cf

siddarthkay committed:
bump okhttp 3.12.12 -> 4.9.1

siddarthkay committed:
remove react-native-codegen

siddarthkay committed:
update .flowconfig for rn66

siddarthkay committed:
update android/app/build.gradle for rn66

siddarthkay committed:
bump react-native 0.65.3 -> 0.66.5

siddarthkay committed:
upgrade react-native 0.66.5->0.67.5

siddarthkay committed:
modify android/build.gradle to support newArch

siddarthkay committed:
adding turbo modules support

siddarthkay committed:
fix incorrect namespace

siddarthkay committed:
upgrading gradle,jdk,buildtools & others

Gradle 6.9 -> 7.5
Gradle Plugin 4.1.0 -> 7.3.0
OpenJDK 8 -> 11
buildToolsVersion 30.0 -> 31.0
react-native-reanimated 2.3.3 -> 2.11.0

siddarthkay committed:
get make run-android to work somehow

Okay so I do get a successful build after doing the following :

updating kotlinPluginVersion from 1.4.3 -> 1.6.20

adding a const in build.gradle for react-native-gesture-handler library to use kotlinPluginVersion

some temp hacks to add https://mvnrepository.com/artifact/com.github.gundy/semver4j to the nix-store

updating deps.json, deps.list and deps.urls by following instructions at https://github.com/status-im/status-mobile/tree/develop/nix/deps/gradle#known-issues, Big thanks to @jakubgs

upgrading react-native-gesture-handler from 2.5.0 -> 2.9.0

removing react-native-mail from the project.

removing react-native-hole-view from the project.

siddarthkay committed:
attempting to get RNHV to work with my fork

siddarthkay committed:
add missing files

siddarthkay committed:
add my fork of react-native-mail

siddarthkay committed:
react-native v0.68.0 -> v0.68.5

flexsurfer commented:
feel free to reopen if needed

siddarthkay commented:
would love to keep this open for experimentation till rn-68 is merged into develop

flexsurfer commented:
@siddarthkay still relevant?

siddarthkay commented:
@flexsurfer : this contains fixes from going to version 0.67 to 0.68 which are relevant

flexsurfer commented:
feel free to reopen if needed ,but only when work is ongoing on it
